### PR TITLE
allow for dynamically providing map provider

### PR DESCRIPTION
**Pull request by docmarionum1**
To allow for providing a custom map provider without having to create a json file in the source directory.

### CODE REVIEW

**Comment by wrobell** [comment]
Why not simply
```
map = Map(..., provider=find_provider(...))
```
?

**Comment by docmarionum1** [comment]
I don't quite understand what you're suggesting.  Are you suggesting that the user, when creating a `Map` object, uses `find_provider`? Changing the default value of `provider` in `Map.__init__`?

File: geotiler/map.py [hunk 1]
```diff
@@ -361,7 +361,7 @@ def render_map(map, downloader=None, loop=None, **kw):
 
 
 @asyncio.coroutine
-def render_map_async(map, downloader=None, loop=None, **kw):
+def render_map_async(map, downloader=None, **kw):
```
Comment (wrobell) [inline geotiler/map.py]: You might be interested in commit 4cc8883.

**Comment by wrobell** [comment]
I have confused myself.

This is *incorrect*
```
map = Map(..., provider=find_provider(...))
```

It should be
```
custom_provider = ...
...
map = Map(...)
map.provider = custom_provider
```

**Comment by pintergreg** [comment]
This would make this lib much more usable, I also came here to open an issue for dynamic providers.
I found a workaround until this is resolved:

```python
carto = {"name": "Carto Positron",
    "attribution": "",
    "url": "https://basemaps.cartocdn.com/rastertiles/light_nolabels/{z}/{x}/{y}.png",
    "subdomains": []
}
carto_provider = geotiler.provider.MapProvider(carto)
carto_tiles = geotiler.Map(extent=bbox, zoom=zoom)  # the default is 'osm', so the provider is OSM at this point
carto_tiles.provider = carto_provider
geotiler.render_map(carto_tiles)
```

The problem is that the constructor only works with string and then uses the the find_provider method the resolve the name to an actual provider read from the JSON files.

Although, the provider can be overwritten by a manually created one, this is not really elegant.

**Comment by wrobell** [comment]
I have merged this pull request. This is possible in ver. 0.14.5 now.

**Pull request merged**